src/udf/src/external.rs

```diff
@@ -59,7 +59,7 @@ impl ArrowFlightUdfClient {
         let full_schema = Schema::try_from(info)
             .map_err(|e| FlightError::DecodeError(format!("Error decoding schema: {e}")))?;
         if input_num > full_schema.fields.len() {
-            return Err(Error::ServiceError(format!(
+            return Err(Error::service_error(format!(
```

Comment:
Perhaps we can keep the name unchanged so that it works fully transparent.

Reply:
`Error` is a struct, so to make it transparent we have to name the constructors in CamelCase. 😕 It seems not a good practice.

Reply:
However, enum variants can be used as constructors by default. 🤔
```rust
enum Error {
    A(i32),
    B(i32, i32),
}

fn main() {
    let f: fn(i32) -> Error = Error::A;
    let f: fn(i32, i32) -> Error = Error::B;
}
```

Reply:
Yeah, but `Error` is a struct instead of enum. 🥵

Reply:
I was always thinking of `Error` as an enum. 🥵 Doesn't matter. The current way LGTM.